File: unit-4/sprint-1/day-1/assignments/RXjava1/app/src/main/java/com/example/rxjava1/MainActivity.java
package com.example.rxjava1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.SystemClock;
import android.util.Log;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.annotations.NonNull;
import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.core.Observer;
import io.reactivex.rxjava3.disposables.Disposable;
import io.reactivex.rxjava3.functions.Predicate;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class MainActivity extends AppCompatActivity {

    private TextView textView;
    private Disposable disposable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textview);
        Task task = new Task(1, "Go to Gym", true);
        Observable<Task> integerObservable = Observable.
                fromIterable(getTaskList())
                .filter(new Predicate<Task>() {
                    @Override
                    public boolean test(Task task) throws Throwable {
                        SystemClock.sleep(1000);
                        return task.isBoo();
                    }
                }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());

        Observer<Task> integerObserver = new Observer<Task>() {
            @Override
            public void onSubscribe(@NonNull Disposable d) {
                disposable = d;
                Log.d("Lloyd", "onSubscribe");
            }

            @Override
            public void onNext(@NonNull Task task) {
                Log.d("Lloyd", "onNext " + task.getId());
                String data = textView.getText().toString() + task.getName();
                textView.setText(data + "\n");
            }

            @Override
            public void onError(@NonNull Throwable e) {
                Log.d("Lloyd", "onError " + e.getMessage());
            }

            @Override
            public void onComplete() {
                Log.d("Lloyd", "onComplete");
                String data = textView.getText().toString() + "Completed";
                textView.setText(data + "\n");
            }
        };

        integerObservable.subscribe(integerObserver);
    }

    private List<Task> getTaskList() {
        List<Task> taskList = new ArrayList<>();

        for (int i = 0; i < 20; i++) {
            if (i % 2 == 0) {
                Task task = new Task(i, "Task - " + i, true);
                taskList.add(task);
            } else {
                Task task = new Task(i, "Task - " + i, false);
                taskList.add(task);
            }
        }
        return taskList;
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        disposable.dispose();
    }
}

File: unit-4/evaluation/U5eval2/app/src/main/java/com/example/u5eval2/MainActivity.kt
package com.example.u5eval2

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.u5eval2.data.Article
import com.example.u5eval2.data.ResponseModel
import com.example.u5eval2.viewModel.ServiceNetwork

class MainActivity : AppCompatActivity() {

    private lateinit var newsAdapter: NewsAdapter
    private lateinit var recyclerView: RecyclerView
    private val resp=ArrayList<Article>()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recyclerView=findViewById(R.id.recycler)

    }
    private fun apicall(){
        ServiceNetwork.getapi().getposts().subs
    }

    private fun setadaptr(){
        newsAdapter= NewsAdapter(resp)
        val layoutManager=LinearLayoutManager(this)
        recyclerView.layoutManager=layoutManager
        recyclerView.adapter=newsAdapter
    }
}